### PR TITLE
chore: sonar setup on LL repo

### PR DESCRIPTION
**Pull request by themooneer**
### ✅ Checklist

- [ ] `npx changeset` was attached.
- [ ] **Covered by automatic tests.** 
- [ ] **Impact of the changes:** 
  - ...

### 📝 Description

- Simply plug-in SonarQube to LL repo
- Collect unit-tests for only LLM and LLD 
- Create a workflow that will be repsonsible for scanning the codebase everyday at 7PM in order to have the latest Code Report.

### ❓ Context

- **JIRA or GitHub link**: 
https://ledgerhq.atlassian.net/browse/LIVE-11380

---

### 🧐 Checklist for the PR Reviewers

- **The code aligns with the requirements** described in the linked JIRA or GitHub issue.
- **The PR description clearly documents the changes** made and explains any technical trade-offs or design decisions.
- **There are no undocumented trade-offs**, technical debt, or maintainability issues.
- **The PR has been tested** thoroughly, and any potential edge cases have been considered and handled.
- **Any new dependencies** have been justified and documented.
- **Performance** considerations have been taken into account. (changes have been profiled or benchmarked if necessary)

### CODE REVIEW

**Review by KVNLS — APPROVED** [review]
Great!

File: .github/workflows/sonar.yml [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    inputs:
+        ref:
+          description: "Which (branch or tag) you want to perform a sonar scan ?"
+          required: false
+          default: develop
+  push:
+    branches:
+      - develop
+  schedule:
+    - cron: "9 19 * * 1-5" # every working day at 7pm we will do a daily scan on develop
```
Comment (KVNLS) [inline .github/workflows/sonar.yml:14]: Do we really need both?

Comment (themooneer) [reply .github/workflows/sonar.yml:14]: you right, no we don't need both, a schedule is sufficient

Comment (themooneer) [reply .github/workflows/sonar.yml]: Done ☑️ thanks

**Review by KVNLS — APPROVED** [review]

**Pull request merged**